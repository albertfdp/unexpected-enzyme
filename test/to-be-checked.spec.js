import expect from './unexpected-enzyme';
import enzyme from 'enzyme';
import React, { Component } from 'react';

class Fixture extends Component {
  render() {
    return (
      <div>
        <input id="checked" defaultChecked />
        <input id="not" defaultChecked={false} />
      </div>
    );
  }
}

describe('to-be-checked', () => {
  ['mount', 'shallow'].forEach(type => {
    let render;

    beforeEach(() => {
      render = enzyme[type];
    });

    describe(type, () => {
      let wrapper;

      beforeEach(() => {
        wrapper = render(<Fixture />);
      });

      it('passes when the actual matches the expected', () => {
        expect(wrapper.find('#checked'), 'to be checked');
      });

      it('passes negated when the actual does not match the expected', () => {
        expect(wrapper.find('#not'), 'not to be checked');
      });

      it('fails when the actual does not match the expected', () => {
        expect(
          () => expect(wrapper.find('#not'), 'to be checked'),
          'with error matching snapshot'
        );

        expect(() => {
          expect(wrapper.find('#checked'), 'not to be checked');
        }, 'with error matching snapshot');
      });
    });
  });
});
