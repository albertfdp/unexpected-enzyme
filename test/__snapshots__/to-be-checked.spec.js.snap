// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to-be-checked mount fails when the actual does not match the expected 1`] = `"expected <input id=\\"not\\"> to be checked"`;

exports[`to-be-checked mount fails when the actual does not match the expected 2`] = `"expected <input id=\\"checked\\"> not to be checked"`;

exports[`to-be-checked shallow fails when the actual does not match the expected 1`] = `"expected <input id=\\"not\\" defaultChecked={false} /> to be checked"`;

exports[`to-be-checked shallow fails when the actual does not match the expected 2`] = `"expected <input id=\\"checked\\" defaultChecked /> not to be checked"`;
